src/types/GBIFTaxon.jl

```diff
@@ -0,0 +1,42 @@
+struct GBIFTaxon
```

Comment:
I feel like there should be a Taxon abstact type, somewhere like in EcoBase, that would have all the fields defined below in lines [2; 3; 6:12]. If all bio packages could agree on a single common identifier (almost like a uuid or doi) for species, it would be very straightforward for packages to link things like occurrences and phylogenies etc via that common base identifier. The packages would not even have to talk together. @richardreeve

Reply:
Oh, wait, that's a TSN.

Reply:
I certainly don't have any objection to an abstract type in EcoBase, but not everyone needs exactly the same list of taxon types - e.g. presence of sub-order, sub-species, absence of phylum, kingdom, etc. - so may need to be quite abstract unlike this specific implementation for GBIF records. PS Not sure what TSN is?

Reply:
Sure thing. A TSN is a Taxonomic serial Number https://www.itis.gov/pdf/faq_itis_tsn.pdf

Reply:
TSN is a taxonomic serial number, which is a uuid for taxa (basically), mapping onto names.

I think having an abstract type defined as `Taxon`, and other types belong to it would be good. Instead of expecting all package-specific types to share the information, we could define something like the `iterate` protocol. This allows package-level info + a general interface. This is starting to sound like an issue for `EcoBase`.

Reply:
yes